scripts/main.js

```diff
@@ -33,63 +33,103 @@ module.exports = (robot) => {
         }
     })
     
-    robot.hear(/bot create milestone|bot close milestone/i, (res) => {
-        var message = res.message.text
+    robot.hear(/bot create milestone/i, (res) => {
+        let message = res.message.text
         message = message.split(' ')
-        if(message.length === 7 && message[0] === "bot" && message[1] === "create" && message[2] === "milestone") {
-          var repos = message[6]
-          repos = repos.split('/')
-          var title = message[3]+" "+message[4]+" "+message[5]
-          const myDate = moment(title, 'DD-MMM-YYYY').toDate()
-          if(myDate == "Invalid Date") {
-            res.reply("Title format is invalid")
-          }
-          else {
-            createMilestone(repos[0], repos[1], title, myDate)
-            res.reply("Milestone successfully created at 'https://github.com/"+repos[0]+"/"+repos[1]+"/milestones")
-          }
-        }
-        else if(message.length === 7 && message[0] === "bot" && message[1] === "close" && message[2] === "milestone") {
-          var repos = message[6]
-          repos = repos.split('/')
-          var title = message[3]+" "+message[4]+" "+message[5]
-          title = "Sprint - "+title
-          const myDate = moment(title, 'DD-MMM-YYYY').toDate()
-          if(myDate == "Invalid Date") {
-            res.reply("Title format is invalid")
-          }
-          else {
-            closeMilestone(repos[0], repos[1], title)
-            res.reply("Milestone successfully closed at 'https://github.com/"+repos[0]+"/"+repos[1]+"/milestones")
-          }
-        }
-        
-        else if(message.length === 6 && message[0] === "bot" && message[1] === "close" && message[2] === "milestone") {
-          var title = message[3]+" "+message[4]+" "+message[5]
-          title = "Sprint - "+title
-          const myDate = moment(title, 'DD-MMM-YYYY').toDate()
-          if(myDate == "Invalid Date") {
-            res.reply("Title format is invalid")
-          }
-          else {
-            closeMilestoneAll(title)
-            res.reply("Milestones successfully closed")
-          }
-        }
-        else if(message.length === 6 && message[0] === "bot" && message[1] === "create" && message[2] === "milestone"){
-          var title = message[3]+" "+message[4]+" "+message[5]
-          const myDate = moment(title, 'DD-MMM-YYYY').toDate()
-          if(myDate == "Invalid Date") {
-            res.reply("Title format is invalid")
-          }
-          else {
-            createMilestoneAll(title, myDate)
-            res.reply("Milestones successfully created")
+        if(message[0] === "bot" && message[1] === "create" && message[2] === "milestone") {
+          switch(message.length) {
+            case 7: 
```

Comment:
I don't think this should be depended on the length of the message or exact/predefined titles for the backlog. I I should be able to give any title to the milestone I want.

Reply:
@zelima if we do not check length of messages, it will hear from the middle of the sentence. Plus, having particular milestones makes less problem. From our milestone history, we usually use `DD MMM YYYY`, `backlog`, `icebox`